CU-10 Gestionar eventos

raunavcru opened the pull request.

angbongon commented:
En las views usas una variable success_url que siempre tiene el mismo valor en todas las CBVs defínela como constante (i.e: SUCCESS_URL = ' ')

angbongon commented:
Además estás creando CBVs en services.py, eso no es correcto, ahí sólo debes crear código destinado a la lógica. Te pongo un ejemplo:
```
class EnrollmentService():
    def is_pending(enrollment_pk: int) -> bool:
        enrollment = models.Enrollment.objects.get(pk=enrollment_pk)
        return enrollment.status == 'PENDING'

    def host_can_update(host: User, enrollment_pk: int) -> bool:
        created_by = models.Enrollment.objects.get(
            pk=enrollment_pk).event.created_by
        return host == created_by


class EventService():
    def enroll(attendee: User, event_pk: int):
        event = models.Event.objects.get(pk=event_pk)

        if attendee.attendee_events.filter(pk=event_pk).exists():
            raise PermissionDenied('Already enrolled')
        attendee.attendee_events.add(event)
        return event
```

Luego puedes importar los servicios en tus vistas (`from . import services`) y después llamar a los servicios y luego a sus funciones.

angbongon commented:
También veo que tienes código duplicado en services y views. Revísalo y vuelve a hacer commit (git push origin nombre de rama) a esta rama, no hagas una nueva rama para corregir los cambios.

angbongon commented on main/templates/main/event_list.html:
```diff
@@ -0,0 +1,21 @@
+  
+
+    <!DOCTYPE html>
+    <head></head>
```
Añade un <title> tag en este html

angbongon commented on main/views.py:
```diff
@@ -1,3 +1,64 @@
+    def form_valid(self, form):
+        object = form.save(commit=False)
+        object.created_by = self.request.user
+        object.created_at = now()
```
Este tipo de cosas como created_at o updated_at la puedes meter en los services y crear una función para ello. Así es reusable

raunavcru committed:
[feat] Implementado el RNF 21 en la parte de crear un evento

[feat] Se ha cambiado las redirecciones al index

[feat] Implementado el calculo de la comisión

[fix] [style] Arreglo de bug y cambio de nombre de las variables

[fix] Se ha convertido los dos métidos en uno

[fix] Modificado el form para que la información extra no sea requerida

angbongon reviewed (changes requested):
Está todo correcto @raunavcru pero hace falta que la información adicional sea opcional, no requerida. Puedes hacerlo desde el objeto formulario poniendolo como required=False. En la rama dev tienes un ejemplo con RegistrationForm.

angbongon commented:
Hay un conflicto como puedes ver, resuelvelo y mergeo la rama @raunavcru

angbongon reviewed (approved):
Todo en orden
